tests/utils/test_config.py

```diff
@@ -215,6 +218,48 @@ def test_internal_type(self):
         cf = self.config_cls.create_from_dict(self.example_dict)
         self.assertTrue(isinstance(cf["deep"], self.config_cls))
 
+    @unittest.skipUnless(
+        check_for_no_backend(),
+        "Test should only be executed if no backend is specified")
+    def test_create_argparser(self):
+        cf = self.config_cls.create_from_dict(self.example_dict)
+        testargs = [
+            '--shallowNum',
+            '10',
+            '--deep.deepStr',
+            'check',
+            '--testlist',
+            'ele1',
+            'ele2',
+            '--setflag']
+        parser = cf.create_argparser()
+        known, unknown = parser.parse_known_args(testargs)
+        self.assertEqual(vars(known)['shallowNum'], 10)
+        self.assertEqual(vars(known)['deep.deepStr'], 'check')
+        self.assertEqual(unknown, ['--testlist', 'ele1', 'ele2', '--setflag'])
+
+    @unittest.skipUnless(
+        check_for_no_backend(),
+        "Test should only be executed if no backend is specified")
+    def test_update_from_argparse(self):
+        cf = self.config_cls.create_from_dict(self.example_dict)
+        testargs = ['--shallowNum', '10',
+                    '--deep.deepStr', 'check',
+                    '--testlist', 'ele1', 'ele2',
+                    '--setflag']
+        # placeholder pyfile because argparser omits first argument from sys
+        # argv
+        with patch.object(sys, 'argv', ['pyfile.py'] + testargs):
+            cf.update_from_argparse(add_unknown_items=True)
+        self.assertEqual(cf['shallowNum'], int(testargs[1]))
+        self.assertEqual(cf['deep']['deepStr'], testargs[3])
+        self.assertEqual(cf['testlist'], testargs[5:7])
+        self.assertEqual(cf['setflag'], True)
+        with warnings.catch_warnings(record=True) as w:
```

Comment:
NIT: it would be better to use [assertWarns](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertWarns) for this, but this way it should also work.

Reply:
thanks for the review, I would suggest merging as is